problems/2721_execute_async_functions_in_parallel/async_functions_in_parallel.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * @param {Array<Function>} functions
+ * @return {Promise<any>}
+ */
+
+const promiseAll = function(functions) {
+    let counter = 0;
+    const results = [];
+    return new Promise((resolve, reject) => {
+        functions.forEach((el, i) => {
+            try {
+                el().then(
+                    result => {
+                        results[i] = result;
+                        counter += 1;
+                        if (counter === [...functions].length) resolve(results);
+                    });
+            } catch (err) {
+                reject(err);
+            }
```

Comment:
тут `try catch` не сработает – у нас асинхронная операция

я вот что имел ввиду:
```suggestion
            el().then(
                result => {
                    results[i] = result;
                    counter += 1;
                    if (counter === [...functions].length) resolve(results);
                }).catch(reject);
```